test/test_nn.py

```diff
@@ -1770,6 +1760,17 @@ def test_Conv2d_inconsistent_types_on_GPU_with_cudnn(self):
         # but it should work with the same type
         nn.functional.conv2d(inputs.float(), weights.float(), bias.float())
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    @unittest.skipIf(not TEST_CUDNN, 'CUDNN not available')
+    def test_Conv2d_deterministic_cudnn(self):
+        inputs = Variable(torch.randn(4, 1, 7, 7).float().cuda())
+        weights = Variable(torch.randn(1, 1, 3, 3).float().cuda())
+        bias = Variable(torch.randn(1).float().cuda())
+        with cudnn.flags(enabled=True, benchmark=True, deterministic=True):
+            f1 = nn.functional.conv2d(inputs, weights, bias)
+            f2 = nn.functional.conv2d(inputs, weights, bias)
+            self.assertEqual(f1, f2)
```

Comment:
The important part of deterministic is also backward part (e.g. for MaxPool). We should check that the grads are equal.

Additionally, non-determinism is often visible at the very small scale (think 1e-6), and `self.assertEqual` compares the tensors pointwise and accepts them as long as no element differs by more than 1e-4. Add a third `0` argument, to force bitwise equality.

Reply:
This particular PR is about conv and not the MaxPool layer
`self.assertEqual` compares tensors and accepts then for 1e-5 but I'll add the third argument 0 as well for the bitwise equality

Reply:
I know, I just don't know the places in conv algos that can cause them to be nondeterministic, while I know the parts that can make MaxPool backward non-deterministic. That's why I used it as an example.